web/source/keymanweb.js

```diff
@@ -3525,22 +3535,106 @@
-       
-    // Add event listeners and attach manually-positioned KMW objects
-    if(keymanweb.options['attachType'] != 'manual')  // I1961
-    {
-      if(document.attachEvent)
-        document.attachEvent('onfocusin', keymanweb._IEFocusIn);
-      else if(document.addEventListener)
-        document.addEventListener('DOMNodeInserted', keymanweb._DOMNodeInserted, true);
-    }
    
     // Restore and reload the currently selected keyboard 
     keymanweb.restoreCurrentKeyboard(); 
 
+      /* Setup of handlers for dynamically-added and (eventually) dynamically-removed elements.
+       * Reference: https://developer.mozilla.org/en/docs/Web/API/MutationObserver
+       * 
+       * We place it here so that it loads after most of the other UI loads, reducing the MutationObserver's overhead.
+       */
+
+      var observationTarget = document.querySelector('body');
+      var k; // our other iteration variables have already been declared previously.
+
+      keymanweb.mutationObserver = new MutationObserver(function(mutations) {
+          var dirtyFlag = false; // Notes if we need to recompute our .sortedInputs array.
+
+          for(var i=0; i < mutations.length; i++) {
+            var mutation = mutations[i];
+            
+            for(var j=0; j < mutation.addedNodes.length; j++) {
+              var addedNode = mutation.addedNodes[j];
+              var lcTagName = "";
+              
+              if(addedNode.tagName) {
+                lcTagName = addedNode.tagName.toLowerCase();
+              }
+              
+              var childAdditions = [];
+              var arrayFromNodeList = function(nl) {
+                var res = [];
+                for(var i=0; i < nl.length; i++) {
+                  res.push(nl[i]);
+                }
+                return res;
+              }
+
+              // Will need to handle this in case of child elements in a newly-added element with child elements.
+              if(addedNode.getElementsByTagName) {
+                childAdditions = childAdditions.concat(
+                  arrayFromNodeList(addedNode.getElementsByTagName('input')),
+                  arrayFromNodeList(addedNode.getElementsByTagName('textarea')),
+                  arrayFromNodeList(addedNode.getElementsByTagName('iframe'))
+                );
+              }
+
+              if(lcTagName == 'input' || lcTagName == 'textarea' || lcTagName == 'iframe') {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(addedNode);
+              }
+
+              for(k = 0; k < childAdditions.length; k++) {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(childAdditions[k]);
+              }
+            }
+
+            // There also exists a 'mutation.removedNodes' array.  We'll need to address that for issue #63.
+
+            // After all mutations have been handled, we need to recompile our .sortedInputs array.
+            if(dirtyFlag) {
```

Comment:
This also moves outside the `for(var i)` and can just be `if(childAdditions.length)` eliminating the need for `dirtyFlag` altogether.